baseplate/diagnostics/tracing/__init__.py

```diff
@@ -494,3 +505,42 @@ def flush_func(self, spans):
             )
         except RequestException as e:
             self.logger.error("Error flushing spans: %s", e)
+
+
+class TraceTooLargeError(Exception):
+    pass
+
+
+class TraceQueueFullError(Exception):
+    pass
+
+
+MAX_SIDECAR_QUEUE_SIZE = 102400
+MAX_SIDECAR_MESSAGE_SIZE = 10000
+
+
+class SidecarRecorder(BaseBatchRecorder):
+    """Interface for recording spans to a POSIX message queue.
+
+    The SidecarRecorder deserializes spans to a string representation before
+    adding them to the queue.
+    """
+    def __init__(self, queue_name):
+        self.queue = MessageQueue(
+            "/traces-" + queue_name,
+            max_messages=MAX_SIDECAR_QUEUE_SIZE,
+            max_message_size=MAX_SIDECAR_MESSAGE_SIZE
+        )
+
+    def send(self, span):
+        # Don't raise exceptions from here. This is called in the
+        # request/response path and should finish cleanly.
+        serialized_str = json.dumps(span._serialize())
+        if len(serialized_str) > MAX_SIDECAR_MESSAGE_SIZE:
+            logger.error("Trace too big")
```

Comment:
maybe include the queue name, observed size, and limit?

Reply:
also, what do we expect application devs to do about this? can we put remediation in the message?

Reply:
Unless MAX_SIDECAR_MESSAGE_SIZE is turned into a configurable value, there's not really anything a dev can do.

I'm fine turning into a configurable but if we do, I propose we do the same to the event publisher in a separate PR.

Reply:
We probably want the limit to stay around -- just boosting it blindly is likely to cause issues. But usually a single span being super large would indicate that it has way too many tags right? So we could say something like "try sending fewer / smaller tags" ?

Reply:
Child spans? yeah that's fair. We ran into a similar situation where thing service was hitting the maximum packet size for metrics and the root cause was that there was no limit on batching for fetching messages and r2 was happily requesting 10k messages in one thrift call which had all sorts of other issues. So maybe just a general admonishment to investigate why the request is generating such a large trace?